feat(VectorStorePGVector Node): Add option to skip table validation and creation (no-changelog)

rhony08 opened the pull request:
## Summary

This update introduces an option to skip table validation and automatic creation (CREATE TABLE IF NOT EXISTS) specifically for the PGVector Store integration. Currently, the default behavior always attempts to create tables if they do not exist, which relies on functionality from the underlying LangChain dependency.

Since LangChain itself supports a flag to skip table validation and creation, this change brings the same flexibility to n8n users. This is important for production environments where database permissions are often segregated between DDL (schema changes) and DML (data manipulation). Many applications restrict AI agents or service accounts to only data read/write operations without permission to create or modify schemas.

By allowing this option, users can prevent runtime errors caused by insufficient permissions when the app needs only to fetch or insert data, improving stability and compatibility with stricter database access policies.

## Related Linear tickets, Github issues, and Community forum posts

<!--
Include links to **Linear ticket** or Github issue or Community forum post.
Important in order to close *automatically* and provide context to reviewers.
https://linear.app/n8n/issue/
-->
<!-- Use "closes #<issue-number>", "fixes #<issue-number>", or "resolves #<issue-number>" to automatically close issues when the PR is merged. -->
### Tested & Added on insert workflows
<img width="959" height="472" alt="image" src="https://github.com/user-attachments/assets/7c27d1a5-5179-4dd4-868b-6065d14372ef" />

### Tested & Added on fetch and load workflows
<img width="959" height="476" alt="image" src="https://github.com/user-attachments/assets/c335e22f-8c96-4cb8-8cc2-9f0c9166c9ca" />



## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

rhony08 committed:
feat(PGVector Store): Add option to skip table validation and creation (no-changelog)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=18640) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=18640) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=18640) it.</sub>

rhony08 committed:
feat(PGVector Store): Change Type Casting to Typeguards

Joffcom commented:
Hey @rhony08,

Thank you for your contribution. We appreciate the time and effort you’ve taken to submit this pull request.

Before we can proceed, please ensure the following:
	•	Tests are included for any new functionality, logic changes or bug fixes.
	•	The PR aligns with our contribution guidelines.

Regarding new nodes:
We no longer accept new nodes directly into the core codebase. Instead, we encourage contributors to follow our Community Node Submission Guide to publish nodes independently.

If your node integrates with an AI service that you own or represent, please email nodes@n8n.io and we will be happy to discuss the best approach.

About review timelines:
This PR has been added to our internal tracker as "GHC-3936". While we plan to review it, we are currently unable to provide an exact timeframe. Our goal is to begin reviews within a month, but this may change depending on team priorities. We will reach out when the review begins.

Thank you again for contributing to n8n.

rhony08 committed:
feat(PGVector Store): Add unit test

rhony08 committed:
feat(PGVector Store): Fix lint error

rhony08 commented on packages/@n8n/nodes-langchain/nodes/vector_store/VectorStorePGVector/VectorStorePGVector.node.ts:
```diff
@@ -241,11 +251,17 @@ export class VectorStorePGVector extends createVectorStoreNode<ExtendedPGVectorS
+			'skipInitializationCheck',
+			0,
+			false,
+		) as boolean;
```
@cubic-dev-ai fixed, please re-review

rhony08 committed:
fix(VectorStorePGVector Node): Resolve merge conflicts after createVectorStoreNode relocation

The shared createVectorStoreNode module was moved to @n8n/ai-utilities on master.
Apply the push.apply fix to the new location, add type annotations for
resultData declarations, and update test mock paths to target the new module paths.

rhony08 committed:
fix(VectorStorePGVector Node): Convert test mocks from jest to vitest

The test was written for Jest but nodes-langchain uses vitest. Updated:
- jest.mock → vi.mock, jest.fn → vi.fn, jest.Mock → Mock type from vitest
- Replaced deep-path mocks with package-level mock for @n8n/ai-utilities
- Added logger and logAiEvent to mock context (real logAiEvent needs them)
